### ISSUE TITLE
[Bug]: list name does not show up in the Lists panel

### ISSUE THREAD

**Issue by andypiper** [GeopJr/Tooth#68]
### Describe the bug

When looking at the... list... of Lists on my account, one of them is a blank row (no name displayed). However, clicking on the blank row shows the list and the correct name as expected. 

![Screenshot from 2023-01-30 12-48-34](https://user-images.githubusercontent.com/552452/215482832-7fd724e8-e0fb-41f9-9402-a60ce69013a3.png)
![Screenshot from 2023-01-30 12-48-53](https://user-images.githubusercontent.com/552452/215482906-8c9ae9e8-7fbd-49f3-9e65-2d30773455e2.png)

### Steps To Reproduce

- Open Tooth.
- Open side panel
- Click on Lists
- Empty row showing for one list name
- Click on empty row
- List opens correctly

### Instance Backend

Mastodon

### Operating System

Ubuntu 22.10

### Package

Compiled manually

### Troubleshooting information

flatpak: false
version: main-79c211c (development)
gtk: 4.8.1 (4.8.1)
libadwaita: 1.2.0 (1.2.0)
libsoup: 2.74.3 (2.74.3)

**Comment by andypiper** [comment]
Ah! I suspected it might be that this was due to punctuation (`&`) in the list name, and I just checked the console output...

```text
** Message: 12:56:56.371: Network.vala:50: GET: https://macaw.social/api/v1/lists?limit=12

(dev.geopjr.tooth:1169306): Gtk-WARNING **: 12:56:56.589: Failed to set text 'Gadgets & Makers' from markup due to error parsing markup: Error on line 1: Entity did not end with a semicolon; most likely you used an ampersand character without intending to start an entity — escape ampersand as &amp;
```

Looks like this is specific to the use of unescaped character entities. Maybe the list names need to be normalised before being passed to the list view here.

**Comment by GeopJr** [comment]
Thanks for raising this issue!

It should be fixed as of c71e1cec3cafa1afe4980ed0599e164f5cce9b89 (saving too)!

![Screenshot of the Tooth gtk app in desktop and dark mode on list view. Theres a single list with the title "test1 % & a τεστ αααα" showcasing that special characters get escaped](https://user-images.githubusercontent.com/18014039/215487767-2c3d6d6d-ad9f-425d-92ae-aba9c79a6189.png)
